stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+    } // namespace _Zip
+
+    template <input_range... _ViewTypes>
+        requires _Zip::_Allow_views<_ViewTypes...>
+    template <bool _IsConst>
+    class zip_view<_ViewTypes...>::_Iterator : public _Iterator_category_container<_Iterator<_IsConst>> {
+    private:
+        friend class zip_view<_ViewTypes...>;
+
+        using _Iterator_tuple_type = tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+        constexpr explicit _Iterator(_Iterator_tuple_type _Current) noexcept(
+            is_nothrow_move_constructible_v<_Iterator_tuple_type>)
+            : _M_Current(_STD move(_Current)) {}
+
+        static constexpr bool _Is_nothrow_dereferenceable =
+            (noexcept(*(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())) && ...);
+
+        static constexpr bool _Is_nothrow_preincrementable =
+            (noexcept(++(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())) && ...);
+
+        static constexpr bool _Is_nothrow_predecrementable =
+            (noexcept(--(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())) && ...);
+
+    public:
+        // NOTE: iterator_category is conditionally present based on whether or not it exists as
+        // a type alias in _Iterator_category_container.
+
+        using iterator_concept = conditional_t<_Zip::_All_random_access<_IsConst, _ViewTypes...>,
+            random_access_iterator_tag,
+            conditional_t<_Zip::_All_bidirectional<_IsConst, _ViewTypes...>, bidirectional_iterator_tag,
+                conditional_t<_Zip::_All_forward<_IsConst, _ViewTypes...>, forward_iterator_tag, input_iterator_tag>>>;
+
+        using value_type      = tuple<range_value_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+        using difference_type = common_type_t<range_difference_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+        _Iterator() noexcept(is_nothrow_default_constructible_v<_Iterator_tuple_type>) = default;
+
+        constexpr _Iterator(_Iterator<!_IsConst> _Rhs) noexcept(
+            (is_nothrow_move_constructible_v<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...)) // strengthened
+            requires(_IsConst
+                     && (convertible_to<iterator_t<_ViewTypes>, iterator_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...))
+            : _M_Current(_STD move(_Rhs._M_Current)) {}
+
+        _NODISCARD constexpr auto operator*() const noexcept(_Is_nothrow_dereferenceable) /* strengthened */ {
+            return _Zip::_Tuple_transform([](auto& _Itr) -> decltype(auto) { return *_Itr; }, _M_Current);
+        }
+
+        constexpr _Iterator& operator++() noexcept(_Is_nothrow_preincrementable) /* strengthened */ {
+            _Zip::_Tuple_for_each([](auto& _Itr) { ++_Itr; }, _M_Current);
+            return *this;
+        }
+
+        constexpr void operator++(int) noexcept(_Is_nothrow_preincrementable) /* strengthened */ {
+            ++(*this);

```

Comment:
We conventionally omit parentheses here, saying `++*this;` because there is no possible confusion about operator precedence here. Occurs below.

Reply:
I do tend to use parentheses often with things like the dereference or address-of operators in my own code, even when it's technically unnecessary according to the C++ specifications. I find that, e.g., 
`&(some_struct_instance.some_member_struct.some_other_member)` is easier to mentally parse than the equivalent without the parentheses.

Still, that's just a personal preference of mine, and I'd be more than willing to adapt similar code for this project's style conventions.